paymentsheet/src/main/java/com/stripe/android/customersheet/CustomerSheetViewModel.kt

```diff
@@ -299,11 +325,15 @@ internal class CustomerSheetViewModel @Inject constructor(
                     ?.transformToPaymentMethodCreateParams(paymentMethodSpec)
                 val paymentMethod = requireNotNull(createPaymentMethod(params))
                 attachPaymentMethodToCustomer(paymentMethod)
-            }.onFailure {
+            }.onFailure { throwable ->
```

Comment:
I find our error handling in this file a bit messy. Some methods return `Result<T>`, some methods throw. Can we clean things up so that methods either return a result _or_ handle results (e.g. by updating the view state), but not both?

Reply:
I think this is because some actions are through `CustomerAdapter`, while some actions are done through `StripeRepository`. `CustomerAdapter` actions returns a `CustomerAdapterResult`, while `StripeRepository` actions return either the model, or Result<StripeModel>, depending on whether we refactored it or not.